RGPosadas/LitterRally
# TASK-54: User Profile Backend

This PR closes #189 

**Testing Pre-requisites**
- Since we now have authenticated routes, we are no longer passing `userId` to a lot of our resolver functions. Therefore, you must add the proper header to the GraphQL queries (Ragith shows how to do so in #200). 
- Neither ApolloStudio or GraphQL Playground don't allow image upload as part of their GraphQL queries. I highly suggest downloading/adding as a chrome extension [Altair](https://altair.sirmuel.design/) as it allows you to do so while keeping GraphQL notation.

**HOW TO TEST: Subtask 54.1 Profile picture is stored in the backend**
1. Create a user (will have avatarId: null)
2. Using Altair, use the updateUser query while adding an image (see screenshot below)
    1. If `avatarId: null`, a new Media entry will be created and the image will be stored in your `uploads/` folder
    2. If `avatarId: number`, then the image stored in your `uploads/` folder will be deleted, and will be updated with the new avatar image while keeping the **same mediaId**

**HOW TO TEST: Subtask 54.2 User statistics are fetched from the backend (portfolio, friends, hosting)**
1. For multiple **different** events, set your user to be:
   1. a participant
   2. a cohost
   3. a host
2. For **some** of those events, set the endDate to be in the past
3. Set your user to have friends
4. You can now test the hostingCount, friendCount, and portfolioCount queries (remember that a user's portfolio includes only events that have now passed)

Below is an example query setup using Altair
![image](https://user-images.githubusercontent.com/33991121/107407735-05ef9680-6ad8-11eb-9dba-e1bba9c747af.png)
